react/components/EditorContainer/StoreEditor/Styles/StyleEditor/StyleEditorHooks.tsx

```diff
@@ -0,0 +1,109 @@
+import { mergeDeepRight } from 'ramda'
+import React, { useReducer, useState } from 'react'
+import { InjectedIntl, injectIntl } from 'react-intl'
+import { ShowToastFunction } from 'vtex.styleguide'
+
+import { RenameStyleFunction } from './mutations/RenameStyle'
+import { UpdateStyleFunction } from './mutations/UpdateStyle'
+import GenerateStyleSheetQuery, {
+  GenerateStyleSheetData,
+} from './queries/GenerateStyleSheet'
+import StyleEditorRouter from './StyleEditorRouter'
+
+interface Props {
+  style: Style
+  intl: InjectedIntl
+  renameStyle: RenameStyleFunction
+  updateStyle: UpdateStyleFunction
+  setStyleAsset: (asset: StyleAssetInfo) => void
+  showToast: ShowToastFunction
+  stopEditing: () => void
+}
+
+type ConfigReducer = (
+  config: TachyonsConfig,
+  partialConfig: Partial<TachyonsConfig>
+) => TachyonsConfig
+
+const StyleEditorStates: React.FunctionComponent<Props> = ({
+  style,
+  intl,
+  renameStyle,
+  setStyleAsset,
+  updateStyle,
+  showToast,
+  stopEditing,
+}) => {
+  const nameState = useState<string>(style.name)
+  const [name] = nameState
+  const configReducer = useReducer<ConfigReducer>(
```

Comment:
What you are doing is basically merging what is dispatched with the current configuration here, right?

Reply:
I think I haven't changed this logic, @pedroteosousa implemented it and I kept as it was - didn't spend much time trying to understand it.